### PR TITLE
 SeatSquares

### PR DESCRIPTION
**Pull request by Khorshid23**
Basic SeatSqures

### CODE REVIEW

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
So, semantically - the best would be to use a checkbox and use the styling aspect on that. Tailwind allows you to have a disabled property on it, and it also accepts a prop called disabled. The other idea would be to make it a button, perhaps easier to style it - and also takes a prop for disabled (reserved)

**Comment by Khorshid23** [comment]
we have converted to a button and we removed the state function since it would be done in the upper layer.

**Comment by ReallyBoard123** [comment]
@TheDoplarEffect why are you making commits on an issue that you are not assigned to?

**Comment by TheDoplarEffect** [comment]
There was a bug on her computer and I was helping her out with the Button typeScript names.
I have assigned myself now as well

File: src/components/SeatSquares.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+type Options = "Available" | "Reserved" | "Selected";
+
+interface ButtonProps
+  extends DetailedHTMLProps<
+    ButtonHTMLAttributes<HTMLButtonElement>,
+    HTMLButtonElement
+  > {
+  square: Options;
+}
+export function SeatSquares({ square, ...props }: ButtonProps) {
```
Comment (ReallyBoard123) [inline src/components/SeatSquares.tsx:12]: This is a single seat, so ideally the naming convention would follow that. SeatSquare or just call it Seat. Squares makes it confusing

File: src/components/SeatSquares.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import clsx from "clsx";
+import { ButtonHTMLAttributes, DetailedHTMLProps } from "react";
+type Options = "Available" | "Reserved" | "Selected";
+
+interface ButtonProps
+  extends DetailedHTMLProps<
+    ButtonHTMLAttributes<HTMLButtonElement>,
+    HTMLButtonElement
+  > {
+  square: Options;
```
Comment (ReallyBoard123) [inline src/components/SeatSquares.tsx:10]: what are these Options?

**Comment by ReallyBoard123** [comment]
@Khorshid23 @TheDoplarEffect, I updated the seat component. Please try it once to see if this is working. Ideally, see if it is taking onClick handler to change the status. I added id props because it might be easier to track which seat to select. Mapping also gets more manageable. Also when adding ...props, make sure to also give it to the component below

**Comment by Khorshid23** [comment]
we changed one of the props from available to reserved and we tested the buttons .I 

https://user-images.githubusercontent.com/116277591/213682339-a12b7fc1-5329-4ec0-91c8-c4cddd7ad40b.mov

would attach the file

**Review by ReallyBoard123 — APPROVED** [review]

**Pull request merged**